modules/core/src/main/java/org/apache/ignite/internal/util/io/GridDataOutput.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.util.io;
+
+import java.io.DataOutput;
+import java.io.IOException;
+import java.io.OutputStream;
+
+/**
+ * Extended data output.
+ */
+public interface GridDataOutput extends DataOutput {
```

Comment:
Maybe we should call it IgniteDataOutput (and IgniteDataInput for Input)? And also GridUnsafe, probably, should be IgniteUnsafe

Reply:
Renamed Grid* to Ignite* (this concerns the classes I brought in this PR). For GridUnsafe, we agreed to do it in a separate PR later.